### DISCUSSION TITLE
Improve external targetability for eggs (like for plugin/mod managers)

### DISCUSSION THREAD

**Discussion by TekExplorer** [pelican-dev/panel#394] [Ideas]
(ie, allow items like plugin/mod managers to easily target eggs without needing to edit the egg manually every time)

So, egg features are pretty cool, but i'd also say they're kinda limited, because it requires that anything that wants to define and use their own, suddenly needs the admins of every panel who use them to edit their eggs.

That or become so wide-spread that people go and edit all eggs in the egg repos.

This is kinda unfeasible.

Instead, I propose some kind of metadata or keyword system.

Hell, we can even keep using the egg features name, but so log as we add much more data for tools to be able to target with.

Information like if it supports forge mods, spigot plugins, _paper_ plugins, if its a proxy, velocity plugins, and so on.

This can allow tools and frontend mods and plugins to better target certain eggs without needing to hard-code a list of egg uuids (which i AM glad is actually possible now)

So then, two possibilities. a dedicated metadata field, or adding much more data to the egg features field. Thoughts? Either way, We'd need to update pretty much all known eggs.

_As the developer of the featherdactyl app, this can make it much easier for me to enable new features in-app without needing to bug the admins to edit their eggs_

**Comment by rmartinoscar** [comment]
I believe Tags can take care of that

**Reply by TekExplorer** [reply]
Maybe if people add a bunch of tags to the eggs in the relevant repos, but its not actually designed for this.
Tags are intended as a way to organize servers and eggs.

Additionally, admins would not have any reason to suspect that tags are used as a targeting mechanism, so it would also present a UX problem.

It _could_ work, but if it does, its by accident. Realistically, that's what egg features are _for_

Perhaps if tags were as json, where we could define multiple different _types_ of tags then went ahead and defined them everywhere in existing eggs, but its all a similar problem.

**Reply by rmartinoscar** [reply]
I didn't see it that way, i saw stuff that lead me to think that egg features were getting replaced (maybe not by Tag) because that they were poorly implemented.
